rma/models/stock_warehouse.py

```diff
@@ -0,0 +1,269 @@
+# Copyright (C) 2017-20 ForgeFlow S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+
+from odoo import fields, models
+
+
+class StockWarehouse(models.Model):
+    _inherit = "stock.warehouse"
+
+    lot_rma_id = fields.Many2one(
+        comodel_name="stock.location", string="RMA Location"
+    )  # not readonly to have the possibility to edit location and
+    # propagate to rma rules (add a auto-update when writing this field?)
+    rma_cust_out_type_id = fields.Many2one(
+        comodel_name="stock.picking.type", string="RMA Customer out Type", readonly=True
+    )
+    rma_sup_out_type_id = fields.Many2one(
+        comodel_name="stock.picking.type", string="RMA Supplier out Type", readonly=True
+    )
+    rma_cust_in_type_id = fields.Many2one(
+        comodel_name="stock.picking.type", string="RMA Customer in Type", readonly=True
+    )
+    rma_sup_in_type_id = fields.Many2one(
+        comodel_name="stock.picking.type", string="RMA Supplier in Type", readonly=True
+    )
+    rma_in_this_wh = fields.Boolean(
+        string="RMA in this Warehouse",
+        help="If set, it will create RMA location, picking types and routes "
+        "for this warehouse.",
+    )
+    rma_customer_in_pull_id = fields.Many2one(
+        comodel_name="stock.rule", string="RMA Customer In Rule"
+    )
+    rma_customer_out_pull_id = fields.Many2one(
+        comodel_name="stock.rule", string="RMA Customer Out Rule"
+    )
+    rma_supplier_in_pull_id = fields.Many2one(
+        comodel_name="stock.rule", string="RMA Supplier In Rule"
+    )
+    rma_supplier_out_pull_id = fields.Many2one(
+        comodel_name="stock.rule", string="RMA Supplier Out Rule"
+    )
+
+    def _get_rma_types(self):
+        return [
+            self.rma_cust_out_type_id,
+            self.rma_sup_out_type_id,
+            self.rma_cust_in_type_id,
+            self.rma_sup_in_type_id,
+        ]
+
+    def _rma_types_available(self):
+        self.ensure_one()
+        rma_types = self._get_rma_types()
+        for r_type in rma_types:
+            if not r_type:
+                return False
+        return True
+
+    def write(self, vals):
+        if "rma_in_this_wh" in vals:
```

Comment:
As stock.rules are created from the values of this field, it could be a good idea to inherit the create method and apply the same logic.